src/core/CorePool.ts

```diff
@@ -311,11 +345,64 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): { position: Position; amount0: JSBI; amount1: JSBI } {
-    // TODO
+    let amount0: JSBI = ZERO,
+      amount1: JSBI = ZERO;
+
+    let position: Position = this.getPosition(owner, tickLower, tickUpper);
+
+    if (JSBI.lessThan(liquidityDelta, ZERO)) {
+      const negatedLiquidityDelta = JSBI.multiply(liquidityDelta, NEGATIVE_ONE);
+      assert(
+        JSBI.greaterThanOrEqual(position.liquidity, negatedLiquidityDelta),
+        "Liquidity Underflow"
+      );
+    }
+
+    assert(tickLower > tickUpper, "tickLower upper than tickUpper");
+    assert(tickLower < TickMath.MIN_TICK, "tickLower lower than MIN_TICK");
+    assert(tickUpper > TickMath.MAX_TICK, "tickUpper bigger than MAX_TICK");
```

Comment:
Please reverse the assertion logic at these three lines according to contract implementation and extract them to a private method so that it can be reused on `collect`, then move the tick checks forward to entrance of method `modifyPosition`.

Reply:
My bad, mix the logic of it, will add private func `checkTicks` to handle to tickLower and tickUpper checks